spec/models/order_updater_spec.rb

```diff
@@ -101,49 +101,30 @@
 
   context '#before_save_hook' do
     let(:distributor) { build(:distributor_enterprise) }
-    let(:order) { build(:order, distributor: distributor) }
-    let(:shipment) { build(:shipment) }
-    let(:shipping_rate) do
-      Spree::ShippingRate.new(
-        shipping_method: shipping_method,
-        selected: true
-      )
-    end
+    let(:shipment) { create(:shipment_with, :shipping_method, shipping_method: shipping_method) }
 
     before do
-      shipment.shipping_rates << shipping_rate
-      order.shipments << shipment
+        order.distributor = distributor
+        order.shipments = [shipment]
     end
 
-    context 'when any of the shipping methods doesn\'t require a delivery address' do
-      let(:shipping_method) { build(:base_shipping_method, require_ship_address: false) }
-
-      let(:delivery_shipment) { build(:shipment) }
-      let(:delivery_shipping_rate) do
-        Spree::ShippingRate.new(
-          shipping_method: build(:base_shipping_method, require_ship_address: true),
-          selected: true
-        )
-      end
+    context 'when shipping method is pickup' do
+      let(:shipping_method) { create(:shipping_method_with, :pickup) }
```

Comment:
is there any reason to have a separate shipping method factory other than better readability when using traits?

Reply:
yeah, I could have extended the spree shipping_method factory, but I think it's better if we keep it with a different name. so, not just readability but also separation from spree code (I think it's similar to extending the class instead of using class_eval).